tests/ui/components/navigation/navigationcomponent.js

```diff
@@ -102,3 +112,70 @@ describe('navigation component configuration', () => {
     expect(tab).toMatchObject(expected);
   });
 });
+
+describe('navigation tab links are correct', () => {
+  let COMPONENT_MANAGER;
+
+  const defaultConfig = {
+    container: '#test-component',
+    verticalKey: 'verticalKey',
+    verticalPages: [
+      { verticalKey: 'vertical', url: './vertical', label: 'Vertical' },
+      { verticalKey: 'facets', url: './facets', label: 'Facets' }
+    ]
+  };
+
+  beforeEach(() => {
+    COMPONENT_MANAGER = mockManager({
+      globalStorage: new GlobalStorage(),
+      persistentStorage: new PersistentStorage()
+    });
+
+    COMPONENT_MANAGER.getComponentNamesForComponentTypes = () => [];
+  });
+
+  it('tab links contain the referrerPageUrl from global storage', () => {
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.REFERRER_PAGE_URL, 'yext.com');
+
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    const navItem = wrapper.find('.js-yxt-navItem').first();
+    const tabLink = navItem.prop('href');
+
+    expect(tabLink).toContain('referrerPageUrl=yext.com');
+  });
+
+  it('tab links contain the context from global storage', () => {
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.API_CONTEXT, 'some context');
+
+    const component = COMPONENT_MANAGER.create('Navigation', defaultConfig);
+    const wrapper = mount(component);
+
+    const navItem = wrapper.find('.js-yxt-navItem').first();
+    const tabLink = navItem.prop('href');
+
+    expect(tabLink).toContain('context=some+context');
+  });
+
+  it('tab links default to the tab order from global storage', () => {
+    const verticalPagesConfig = new VerticalPagesConfig([
+      { label: 'Home', url: './index.html' },
+      { label: 'People', url: './people.html', verticalKey: 'people' }
+    ]);
+
+    COMPONENT_MANAGER.core.globalStorage.set(StorageKeys.VERTICAL_PAGES_CONFIG, verticalPagesConfig);
+
+    const component = COMPONENT_MANAGER.create('Navigation', {
+      container: '#test-component',
+      verticalKey: 'verticalKey',
+      verticalPages: null
+    });
+    const wrapper = mount(component);
```

Comment:
How slow is the component creation and `mount`? If performance isn't great, we can consolidate things by setting `globalStorage` and creating the component once. Then, we'd have three tests, one to verify each property was present in the URL. If performance isn't a problem, no worries.

Reply:
The time spent on component creation and mount seems negligible. We do that in lots of test (about 60), and our tests are relatively quick (about 6-7 seconds for the whole test suite)